files/stat.py

```diff
@@ -469,15 +518,30 @@ def main():
     # try to get mime data if requested
     if get_mime:
         output['mimetype'] = output['charset'] = 'unknown'
-        filecmd = [module.get_bin_path('file', True), '-i', path]
-        try:
-            rc, out, err = module.run_command(filecmd)
-            if rc == 0:
-                mimetype, charset = out.split(':')[1].split(';')
-                output['mimetype'] = mimetype.strip()
-                output['charset'] = charset.split('=')[1].strip()
-        except:
-            pass
+        filecmd = [module.get_bin_path('file'), '-i', path]
+        if filecmd:
+            try:
+                rc, out, err = module.run_command(filecmd)
+                if rc == 0:
+                    mimetype, charset = out.split(':')[1].split(';')
+                    output['mimetype'] = mimetype.strip()
+                    output['charset'] = charset.split('=')[1].strip()
+            except:
+                pass
```

Comment:
It feels like we should change the default value of mime to True if we do this.  Right now the UI is:
* User explicitly requests that we get mime information for this file.  If we can't do that, because file is unavailable then fail. (Bug: if we can't do that because file fails then we should also fail).

If we make your change but don't change the default it reads as:
* User explicitly asked for mime information about this file.  Maybe we'll give that to them or maybe we won't.

If we make your change and change the default it then reads:
* Normally we get the mime information iif we can.  If the user explicitly says not to then we don't bother.

I like the last one the best.  The first one is my second choice (and we should fix non-zero exit to also fail_json if we do that)